instance/models/instance.py

```diff
@@ -17,656 +17,112 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 """
-Instance app models - Instance
+Instance app models - Open EdX Instance and AppServer models
 """
-
-# Imports #####################################################################
-
 import logging
-import string
 
-from django.conf import settings
+from django.contrib.contenttypes.fields import GenericForeignKey, GenericRelation
+from django.contrib.contenttypes.models import ContentType
 from django.db import models
-from django.db.backends.utils import truncate_name
-from django.template import loader
-from django.utils import timezone
-from django.utils.crypto import get_random_string
+from django.utils.functional import cached_property
 from django_extensions.db.models import TimeStampedModel
-from functools import partial
 
-from instance.gandi import GandiAPI
 from instance.logger_adapter import InstanceLoggerAdapter
-from instance.logging import log_exception
-from instance.models.mixins.ansible import AnsibleInstanceMixin
-from instance.models.mixins.database import MongoDBInstanceMixin, MySQLInstanceMixin, SwiftContainerInstanceMixin
-from instance.models.mixins.utilities import EmailInstanceMixin
-from instance.models.mixins.version_control import GitHubInstanceMixin
-from instance.models.utils import (
-    ModelResourceStateDescriptor, ResourceState, SteadyStateException, ValidateModelMixin
-)
-
-# Constants ###################################################################
-
-PROTOCOL_CHOICES = (
-    ('http', 'HTTP - Unencrypted clear text'),
-    ('https', 'HTTPS - Encrypted'),
-)
-
-gandi = GandiAPI()
+from .utils import ValidateModelMixin
```

Comment:
@bradenmacdonald I don't have a strong objection to them, but I kept away from them for consistency -- I would still prefer to keep it that way, but won't oppose using them if you feel strongly about it. 

For your remark, note that when all imports are absolute, there is usually no ambiguity as to what `instance` we are referring to in an import, as it will always be the base module.